scripts/zones/Ifrits_Cauldron/mobs/Bomb_Queen.lua

```diff
@@ -1,35 +1,53 @@
 -----------------------------------
 -- Area: Ifrit's Cauldron
 --   NM: Bomb Queen
+--  Vid: https://www.youtube.com/watch?v=AVsEbYjSAHM
 -----------------------------------
 local ID = require("scripts/zones/Ifrits_Cauldron/IDs")
 require("scripts/globals/status")
 -----------------------------------
 
 function onMobInitialize(mob)
+    mob:setMobMod(dsp.mobMod.IDLE_DESPAWN, 180)
     mob:setMobMod(dsp.mobMod.HP_STANDBACK, -1)
     mob:setMobMod(dsp.mobMod.DRAW_IN, 1)
-    mob:setMobMod(dsp.mobMod.IDLE_DESPAWN, 180)
+    mob:setMod(dsp.mod.STUNRES, 50)
 end
 
 function onMobSpawn(mob)
-    mob:addMod(dsp.mod.STUNRES, 50)
+    mob:setLocalVar("petCooldown", os.time() + 5) -- five seconds for first pet
 end
 
-function onMobFight(mob,target)
-    local battleTime = mob:getBattleTime()
+function onMobFight(mob, target)
+    -- every ~20 seconds after first pet, a new pet will spawn around queen's position.
+    -- 50% chance for Prince/Princess. 1% chance for Bastard.
 
-    if (battleTime - 5) % 20 == 0 then -- every 20 seconds after 5 seconds, source: https://www.youtube.com/watch?v=AVsEbYjSAHM
-        local random = math.random(1,100);
-        local petId = mob:getID() + 1 + math.floor(random / 25); -- approximately 1% chance to spawn bomb bastard
+    local mobId = mob:getID()
+
+    if os.time() > mob:getLocalVar("petCooldown") then
+        local petId = mob:getID() + 1 + math.floor(math.random(0, 100) / 25)
```

Comment:
0  to 100 technically makes it a 50/101 chance for the prince

Reply:
That would make the minimum chance 4%, which is too high for bomb bastard. The best implementation would probably be `math.floor(math.random(0, 199) / 49)`, which would give a 24.5% chance for each prince, each princess, and 1% for bastard.